GOVERNANCE.md

```diff
@@ -4,83 +4,139 @@ This document defines project governance for the Strimzi project.
+
 ### Maintainers
 
-The maintainers are responsible for the overall development of the whole Strimzi project and all its components.
-The maintainers are identified in the [`MAINTAINERS`](MAINTAINERS) file. 
+A maintainer is actively responsible for the overall development of the whole Strimzi project and all its components.
+The maintainers are identified in the [`MAINTAINERS`](MAINTAINERS) file.
 
-#### Changes in Maintainership
+Every maintainer is, in effect, also an owner for all components and does not have to be mentioned in the `COMPONENT-OWNERS` file.
 
-New maintainers are proposed by an existing maintainer and are elected by a ⅔ majority maintainers vote.
-Maintainers can be removed by a ⅔ majority maintainers vote.
+### Emeritus code owners and maintainers
 
-#### Github Project Administration
+An emeritus code owner, or an emeritus maintainer, is someone who is formerly had that role but is no longer actively involved with the project.
+Emeritus code owners and Emeritus maintainers are identified in the [`EMERITUS`](EMERITUS) file.
 
-Maintainers will be added to the collaborators list of the Strimzi repository with "Write" access.
-After 6 months a maintainer will be given "Admin" access to the Strimzi repository.
+In the rest of this document the term 'maintainer' should be taken to mean 'active maintainer' unless qualified using the term 'emeritus maintainer'.
 
-### Component owners
+### Changes in Maintainership
 
-The component owners are responsible for development of a specific subproject or component within Strimzi.
-Such components might be represented by a separate GitHub repository within the Strimzi organization (e.g. `strimzi-kafka-bridge`) or by a subdirectory in a GitHub repository (e.g. `./documentation/` in `strimzi-kafka-operator`).
-The components owners are identified in the [`COMPONENT-OWNERS`](COMPONENT-OWNERS) file in this repository which also includes the component for which they are responsible.
+New maintainers are proposed by an existing maintainer and are elected by a vote (see later for the rules of this vote).
+If the vote succeeds the proposed maintainer will be invited to become a maintainer.
+If the proposed maintainer accepts the invitation:
+* Their name will be added to the `MAINTAINERS` file.
+* They will be added to the collaborators list of the Strimzi repository with "Write" access
+* After 6 months they will be given "Admin" access to the Strimzi repository.
+
+Maintainers can be removed by a vote.
+If the vote succeeds:
+* Their name will be removed from the `MAINTAINERS` file.
+* They will be removed to the collaborators list of the Strimzi repository
 
-By definition, every maintainer is also an owner for all components and does not have to be mentioned in the `OWNERS` list.
+Maintainers who wish to become emeritus simply mark the fact by opening a PR against this repository moving their name to the `EMERITUS` file.
```

Comment:
So it means that if an inactive maintainer doesn't open any PR (because he/she is just inactive, not following the project, and not having any interest to open any kind of PR on it), it will be forever maintainer even if not active?
Isn't there any "inactivity" rule to make him/her emeritus or having a PR open by another maintainer?

Reply:
The mechanism I'm proposing for `active -> emeritus` is entirely voluntary.

The change from ⅔ _all_ maintainer approval to ⅔ _voting_ maintainer approval means that such inactive maintainers don't prevent decisions being taken due to their inactivity (which is a problem we want to solve, even though it's not yet been a problem in practice). 

Obviously that doesn't prevent someone who hasn't been active for a long time suddenly voting on some contentious issue and swinging a decision that otherwise would have gone the other way. TBH that seems really unlikely: we certainly don't have a problem with rogue maintainers currently. And there is (and always was) a mechanism for the other maintainers to remove a maintainer. So if someone did this then the other maintainers could get rid of them anyway.

Reply:
Why do you think inactivity is not a good way anymore? Also, you said "there is (and always was) a mechanism for the other maintainers to remove a maintainer." ... but it's not about making someone "emeritus", it's about removing it. Why can't we have a mechanism where a maintainer can propose another one is inactive enough to be emeritus? I think if we have just "active -> emeritus is entirely voluntary.", the inactive maintainers will be just there forever (because they are inactive and not following at all what's happening here, so even not reading this proposal and not marking themselves as emeritus) ... and it won't reflect the reality of the maintainers pool.

Reply:
I would not have a problem with a maintainer being able to propose to move someone to emeritus. But in a situation when it is not needed (i.e. after the ⅔ of all maintainer vote is changed), I would take it as a pretty aggressive move to be honest.

Reply:
I was the object of such a thing on the Eclipse Hono project and I found it pretty fair. I was not involved anymore as before and the project lead sent out an email to propose me to move to be "historical committer" (as you can see here https://projects.eclipse.org/projects/iot.hono/who). I just accepted because it was reflecting the reality.

Reply:
> Why do you think inactivity is not a good way anymore?

It just seems unnecessary to invent a whole set of extra rules and processes. By only counting the votes of people who are actively participating (in the vote) we prevent the problem we were trying to solve, but with fewer rules.

> Why can't we have a mechanism where a maintainer can propose another one is inactive enough to be emeritus? 

Why do we need a codified mechanism to do that? We could just, you know, just ask them: "I see you're no longer active in the project, would you mind opening a PR to reflect that? Or did you intend to start contributing again soon?"

Let me flip the question around: What advantage do you see to having rules to automatically make maintainers emeritus?

Reply:
I would like just to have the reality reflected in the maintainers pool.
Btw asking someone to step back could be even another way as you suggest, not sure about getting a reply btw.